internal/cmd/login.go

```diff
@@ -81,16 +85,32 @@ $ infra login --key 1M4CWy9wF5.fAKeKEy5sMLH9ZZzAur0ZIjy`,
 }
 
 func login(cli *CLI, options loginCmdOptions) error {
-	var err error
+	config, err := readConfig()
+	if err != nil {
+		return err
+	}
 
 	if options.Server == "" {
-		options.Server, err = promptServer(cli, options)
+		if options.NonInteractive {
+			return Error{Message: "Non-interactive login requires the [SERVER] argument"}
+		}
+
+		options.Server, err = promptServer(cli, config)
 		if err != nil {
 			return err
 		}
 	}
 
-	client, err := newAPIClient(cli, options)
+	if len(options.TrustedCertificate) == 0 {
```

Comment:
seems like this will always be true in the current implementation

Reply:
Ya, we need to add a `--tls-trusted-ca` or `--tls-trusted-cert` flag to the CLI. I was thinking of doing that as part of #2176.

There are some TODOs in this PR that mention it as well.

Reply:
I opened #2288 which resolves those TODOs and adds the new flag.